src/lib/plugins/addPagination.ts

```diff
@@ -27,18 +34,27 @@ export const createPageStore = ({ items, initialPageSize, initialPageIndex }: Pa
 	};
 	const setPageSize = (newPageSize: number) => updatePageSize(() => newPageSize);
 
-	const pageCount = derived([pageSize, items], ([$pageSize, $items]) => {
-		const $pageCount = Math.ceil($items.length / $pageSize);
+	const pageIndex = writable(initialPageIndex);
+
+	let pageCount;
```

Comment:
Move this declaration down to the `if` statement on line 52 to make it clearer that it's a conditional assignment.